### PR TITLE
FIX: ICA.plot_properties wont accept new extrapolate argument of viz.…

### PR DESCRIPTION
**Pull request by jscastanoc**
…plot_topomap

#### Reference issue
--

#### What does this implement/fix?
plot_topomap has included recently a "extrapolate" argument (https://github.com/mne-tools/mne-python/pull/5754), however, the _plot_ica_topomap is not accepting this parameter. This pull request fixes this

#### Additional information
---

### CODE REVIEW

**Comment by agramfort** [comment]
@jscastanoc you have CI failure

also you are using directly _plot_ica_topomap? why not using directly the public API?

**Comment by jscastanoc** [comment]
> @jscastanoc you have CI failure
> 
> also you are using directly _plot_ica_topomap? why not using directly the public API?

whoops, gonna check on that. And no, I get the error when calling ICA.plot_properties, that is the function that I am using

File: mne/viz/topomap.py [hunk 1]
```diff
@@ -965,7 +965,7 @@ def _plot_ica_topomap(ica, idx=0, ch_type=None, res=64, layout=None,
                       vmin=None, vmax=None, cmap='RdBu_r', colorbar=False,
                       title=None, show=True, outlines='head', contours=6,
                       image_interp='bilinear', head_pos=None, axes=None,
-                      sensors=True, allow_ref_meg=False):
+                      sensors=True, allow_ref_meg=False, extrapolate='box'):
```
Comment (agramfort) [inline mne/viz/topomap.py]: is extrapolate='box' the default so this does not change any current behavior?

Comment (jscastanoc) [reply mne/viz/topomap.py]: yes: https://github.com/mne-tools/mne-python/blob/master/mne/viz/topomap.py#L628

**Comment by agramfort** [comment]
so we agree it's just for your personal usage that relies on a private
function?

>

**Comment by jscastanoc** [comment]
> so we agree it's just for your personal usage that relies on a private function?
> […](#)

I would not agree that it's for personal use: Without this addition, we could not access the new extrapolate options implemented for topo-plotting (in pull-request [#5754](https://github.com/mne-tools/mne-python/pull/5754)) when plotting ICA patterns (for example when calling ICA.plot_properties). I am not calling _plot_ica_topomap directly.

File: mne/viz/topomap.py (hunk 1 above)
Comment (mmagnuski) [reply mne/viz/topomap.py]: yes, 'box' is the old behavior

**Comment by mmagnuski** [comment]
@agramfort That's good change, I think, not personal use: the extrapolate arg should be available in all topo plotting functions in my opinion. This is something we agreed on in my PR, but I was to check and add support for `extrapolate` in other function in a successive PR. I can take care of this here in the upcoming weekend.

**Comment by agramfort** [comment]
what I don't get is that you only change the API of _plot_ica_topomap function which is private. Am I missing something?

also @jscastanoc you should not open a PR from your master branch.

**Comment by jscastanoc** [comment]
> what I don't get is that you only change the API of _plot_ica_topomap function which is private. Am I missing something?
> 
> also @jscastanoc you should not open a PR from your master branch.

the _plot_topomap implements the "extrapolate" argument, which is accessible from the plot_topomap (this is the PR @mmagnuski is talking about)

However, when calling ICA.plot_properties, plot_topomap gets called via _plot_ica_topomap, ergo if _plot_ica_topomap does not implement extrapolate, this argument cannot be used in plot_topomap:

Currently:
ICA.plot_properties -> _plot_ica_topomap (doesnt implement extrapolate) -> plot_topomap (implements extrapolate but cannot be accessed since the caller _plot_ica_topomap does not implement it)

I hope I could explain myself a bit clearer now.

PS. Sorry about the PR from master, dont have much experience contributing to big projects. I'll keep that in mind for next time.

**Comment by mmagnuski** [comment]
@agramfort That is because `plot_properties` has `topomap_args` which is used to pass additional args to `_plot_ica_topomap` so no need to expose extrapolate in `plot_properties`.

**Comment by agramfort** [comment]
@jscastanoc can you push a tiny commit to restart the CIs. You have spurious failure and I cannot restart azure myself it seems

**Comment by larsoner** [comment]
@jscastanoc can you add an entry to the BUG section of `whats_new.rst` like:
```
- Fix support for supplying ``extrapolate`` via :ref:`plot_properties(..., topomap_args=dict(extrapolate=...)) <mne.preprocessing.ICA.plot_properties>` by `Firstname Lastname`_

...
<and make sure later in the file there is>
.. _Firstname Lastname: https://your-preferred-link-url
```

**Comment by agramfort** [comment]
@jscastanoc can you rebase to make CIs happy?

**Comment by jscastanoc** [comment]
> @jscastanoc can you rebase to make CIs happy?

@agramfort all set, I think :D (apparently some of the tests are failing randomly ?)

**Comment by agramfort** [comment]
@jscastanoc can you rebase again? I don't manage to make circle green despite some restarts

**Comment by jscastanoc** [comment]
@agramfort could you please give me a hint on this? I have no idea why that test is failing

**Comment by mmagnuski** [comment]
Circle falls over when building docs (`make html`), the error is:
```
Makefile:54: recipe for target 'html_dev-noplot' failed
make: *** [html_dev-noplot] Error 1
Exited with code 2
```
I don't know why this happens.

**Comment by larsoner** [comment]
Look at the log, warnings become errors and there is this:
```
/home/circleci/project/doc/whats_new.rst:169: WARNING: undefined label: mne.preprocessing.ica.plot_properties (if the link has no caption the label must precede a section header)
```
I've pushed a commit to fix it

**Comment by agramfort** [comment]
thx @jscastanoc

**Pull request merged**